pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig_test.go

```diff
@@ -0,0 +1,204 @@
+package bootstrapperconfig
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/pkg/webhook"
+	dtclientmock "github.com/Dynatrace/dynatrace-operator/test/mocks/pkg/clients/dynatrace"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testPaasToken       = "test-paas-token"
+	testAPIToken        = "test-api-token"
+	testDataIngestToken = "test-ingest-token"
+
+	testUUID                  = "test-uuid"
+	testTenantToken           = "abcd"
+	testCommunicationEndpoint = "https://tenant.dev.dynatracelabs.com:443"
+
+	testHost = "test-host"
+
+	testDynakube   = "test-dynakube"
+	testNamespace  = "test-namespace"
+	testNamespace2 = "test-namespace2"
+
+	testNamespaceDynatrace = "dynatrace"
+
+	testApiUrl = "https://" + testHost + "/e/" + testUUID + "/api"
+)
+
+func TestNewSecretGenerator(t *testing.T) {
+	client := fake.NewClient()
+	mockDTClient := dtclientmock.NewClient(t)
+
+	secretGenerator := NewSecretGenerator(client, client, mockDTClient)
+	assert.NotNil(t, secretGenerator)
+
+	assert.Equal(t, client, secretGenerator.client)
+	assert.Equal(t, client, secretGenerator.apiReader)
+	assert.Equal(t, mockDTClient, secretGenerator.dtClient)
+}
+
+func TestGenerateForDynakube(t *testing.T) {
+	t.Run("succcessfully generate secret for dynakube", func(t *testing.T) {
+		dynakube := &dynakube.DynaKube{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testDynakube,
+				Namespace: testNamespaceDynatrace,
+			},
+			Spec: dynakube.DynaKubeSpec{
+				APIURL: testApiUrl,
+			},
+		}
+
+		clt := fake.NewClientWithIndex(
+			dynakube,
+			clientInjectedNamespace(testNamespace, testDynakube),
+			clientSecret(testDynakube, testNamespaceDynatrace, map[string][]byte{
+				dtclient.ApiToken:  []byte(testAPIToken),
+				dtclient.PaasToken: []byte(testPaasToken),
+			}),
+			clientSecret(dynakube.OneAgent().GetTenantSecret(), testNamespaceDynatrace, map[string][]byte{
+				"tenant-token": []byte(testTenantToken),
+			}),
+			clientSecret(consts.BootstrapperInitSecretName, testNamespace, nil),
```

Comment:
`secretGenerator.GenerateForDynakube` should create this secret
- it could be a valid test, that it gets updated 👀 

you could also set the certs `trustedCA` and `tlsSecretName` here (and in the DK, so you check every field)

also there might be an interesting test-case:
1. secret already exists with certain optional fields already set (like trustedCAs)
2. the new secret wouldn't have this optional field
3. after the update, will the optional field be present?
  - it shouldn't be
  - this maaay be a bug in the current implementation

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/caf3dcbdf47a53cbf539a588a930e1dc1c10eee8

Well, I can confirm In a test case there the secret already exists. After updating the fields that were already there `trustedCAs´ and `tlsSecretName` are still there but with the updated fields. How should we proceed there? And why would this be a bug?

Reply:
The bug would be that the secret HAS trustedCA stuff, while the Dynakube does NOT

Reply:
Oh wait yeah sure, this would not be expected, but then I did the test wrong, because my DK has those fields. After generating the secret the fields are updated.

What you're saying is that if there happens to exist the secret beforehand with certain fields that are **not** in the Dynakube, and when we then update the secret, the fields should **not** be there anymore, right?

I will update the test to reflect exactly this.

Only question remains: Is this really a bug? I mean the secret is coupled to the Dynakube, it gets deleted as soon as the Dynakube isn't there. 

I guess we want to cover the case if there would fail this mechanism right?

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/4e76d50a8ac0a66765d7bf5660e55877985350d3

updated the test, IMO it works as it should. The values are not there when updating the secret, because they did not exist in the Dynakube. WDYT?

Reply:
excelent, because we set it to `""`, therefore it gets updated
and I verified that if these fields are set to `""` the bootstrapper works
- it seems, k8s doesn't create empty files for these "keys" that have no value